autoscheduler/frontend/src/components/SchedulingPage/ConfigureCard/ConfigureCard.tsx

```diff
@@ -89,7 +107,7 @@ const ConfigureCard: React.FC = () => {
         <div id={styles.cardHeader}>Configure</div>
       }
     >
-      <div className={styles.buttonContainer}>
+      <div className={styles.buttonContainer} ref={ref}>
```

Comment:
Just for my understanding, what does adding this ref do for us?

Reply:
I'll add a comment above the initial `useRef` because you're right that it's not immediately clear what its purpose is. Basically, when I was running tests, I kept getting an error about React dispatching a `setState` in an async callback after un-mounting a component. My solution to this was to obtain a reference to the DOM element and check if it's still mounted before calling any asynchronous `setState()`s.

TL;DR the ref fixes async setState